scripts/run-clang-tidy.py

```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/env python
+# SPDX-License-Identifier: Apache-2.0
+
+# Run clang-tidy recursively and parallel on directory
+# Usage: run-clang-tidy sourcedir builddir excludedirs extensions
+# extensions and excludedirs are specified as comma-separated
+# string without dot, e.g. 'c,cpp'
+# e.g. run-clang-tidy . build test,other c,cpp file
+
+import os, sys, subprocess, multiprocessing
+manager = multiprocessing.Manager()
+failedfiles = manager.list()
+
+# Get absolute current path and remove trailing seperators
+currentdir = os.path.realpath(os.getcwd()).rstrip(os.sep)
+print("Arguments: " + str(sys.argv))
+# Get absolute source dir after removing leading and trailing seperators from input.
+sourcedir = currentdir + sys.argv[1].lstrip(os.sep).rstrip(os.sep)
+print("Source directory: " + sourcedir)
+builddir = sourcedir + os.sep + sys.argv[2].rstrip(os.sep)
+print("Build directory: " + builddir)
+# If exclude dirs is not empty, split it into a tuple
+excludedirs = ()
+if sys.argv[3]:
+    excludedirs = tuple([(sourcedir + os.sep + s).rstrip(os.sep) for s in sys.argv[3].split(',')])
```

Comment:
```suggestion
excludedirs = tuple([(sourcedir + os.sep + s).rstrip(os.sep) for s in sys.argv[3].split(',')])
```

Probably no need for this if check since you are relying on sys.argv[4] being present below.